### DISCUSSION TITLE
How do I stop these branches from getting recreated?

### DISCUSSION THREAD

**Discussion by tubbo** [renovatebot/renovate#14852] [Request Help] [answered]
These two dependency updates are constantly failing our build, but every time I try to close the PR they get re-created. It's very annoying and takes up precious rate limit slots:

![Screen Shot 2022-03-29 at 2 44 41 PM](https://user-images.githubusercontent.com/113026/160682942-9c4e9418-9448-450e-93fd-62f2b6c0a3bb.png)

```json
{
  "timezone": "America/New_York",
  "extends": ["config:js-app", ":semanticCommitTypeAll(chore)"],
  "rebaseWhen": "conflicted",
  "recreateClosed": false,
  "packageRules": [
    {
      "description": "Automatically merge development dependency upgrades when they pass CI",
      "matchDepTypes": ["devDependencies"],
      "automerge": true
    },
    {
      "description": "Automatically merge dependency and digest pins when they pass CI",
      "matchDepTypes": ["pin", "digest"],
      "automerge": true
    },
    {
      "description": "Pin GitHub Actions dependencies",
      "depTypeList": ["action"],
      "extractVersion": "^(?<version>v\\d+\\.\\d+\\.\\d+)$",
      "versioning": "regex:^v(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?$"
    },
    {
      "groupName": "dicebear",
      "matchSourceUrlPrefixes": ["https://github.com/dicebear/dicebear"]
    },
    {
      "groupName": "sendgrid",
      "matchSourceUrlPrefixes": ["https://github.com/sendgrid/sendgrid-nodejs"]
    },
    {
      "groupName": "radix",
      "matchSourceUrlPrefixes": ["https://github.com/radix-ui/primitives"]
    },
    {
      "matchSourceUrlPrefixes": ["https://github.com/sanity-io/sanity"],
      "enabled": false
    }
  ],
  "ignoreDeps": ["@commitlint/cli", "@commitlint/config-conventional"]
}
```

I'm not interested in removing them _forever_, like if a new update comes along that fixes their problems that would be awesome and I would love to merge it, but maybe it would be nice to not annoy me throughout the day as the PR constantly gets re-created every time Renovate checks my repo...

**Comment by rarkins** [comment, marked as answer]
The first one has a version in its title so I wouldn't expect it to be recreated if you close it. In the PR body does it say that it's "immortal"?

For the second one, it will be immortal because there's no version in the PR title (such as "v4"). Can you confirm that there's a mix of major versions in that PR?

Something you could consider is using the dependency dashboard approval feature for major updates, like we even do in this repo. That way you won't get major updates by default, and instead only when you tick the checkbox to create them when you're ready for them.

**Reply by tubbo** [reply]
@rarkins yes, this branch is marked as "immortal" and I kinda want to change it just for this one dependency because it's gridlocked in terms of fixing its problems.

![Screen Shot 2022-04-01 at 3 51 10 PM](https://user-images.githubusercontent.com/113026/161332093-567a1d21-fef6-4451-bce2-470dae232924.png)

**Reply by rarkins** [reply]
Any branch with a mismatch of versions like that will unfortunately be immortal. The only thing you can do is configure dependencyDashboardApproval=true so that it's not created automatically in future

**Reply by tubbo** [reply]
Thanks so much @rarkins!

**Reply by BusyByte** [reply]
@rarkins we are seeing the same thing with it being immortal on even just patch differences:
```html
<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This PR contains the following updates:</p>

Package | Update | Change
-- | -- | --
org.http4s:http4s-blaze-client (source) | minor | 0.21.33 -> 0.23.12
org.http4s:http4s-blaze-server (source) | minor | 0.21.33 -> 0.23.12
org.http4s:http4s-circe (source) | minor | 0.21.33 -> 0.23.16
org.http4s:http4s-client (source) | minor | 0.21.33 -> 0.23.16
org.http4s:http4s-dsl (source) | minor | 0.21.33 -> 0.23.16

<br class="Apple-interchange-newline">This PR contains the following updates:

Package	Update	Change
[org.http4s:http4s-blaze-client](https://http4s.org/) ([source](https://github.com/http4s/http4s))	minor	0.21.33 -> 0.23.12
[org.http4s:http4s-blaze-server](https://http4s.org/) ([source](https://github.com/http4s/http4s))	minor	0.21.33 -> 0.23.12
[org.http4s:http4s-circe](https://http4s.org/) ([source](https://github.com/http4s/http4s))	minor	0.21.33 -> 0.23.16
[org.http4s:http4s-client](https://http4s.org/) ([source](https://github.com/http4s/http4s))	minor	0.21.33 -> 0.23.16
[org.http4s:http4s-dsl](https://http4s.org/) ([source](https://github.com/http4s/http4s))	minor	0.21.33 -> 0.23.16
```

0.23.12 is not even binary compatible with 0.21.33  since it's developer early semantic versioning.
But it's being reopened because it's immortal and I'd like this one to be ignored until there's a new 0.21.x or 0.23.x patch version.

Any advice on this one or are we limited to `dependencyDashboardApproval`? We wouldn't want to do use dependencyDashboardApproval, right now we just added a comment which says do not merge which is not a great solution as the PR keeps getting pushed to and the comments get hidden.

**Reply by rarkins** [reply]
@BusyByte Immortal PRs happen when there is no common "to version" in a grouped PR. You are showing an example where there's no common to version, so an immortal PR is expected. If you have grouped packages without identical versions and wish to ignore them, then dependency dashboard approval is the recommended way

**Reply by BusyByte** [reply]
@rarkins  I think these are grouped by the default config because they all have the same groupId.  It's not anything specific I have configured.  Let me try separating the version identifier val and see if that helps.

**Reply by BusyByte** [reply]
@rarkins I've separated out the vals for the versions and now I get 2 PRs which are not immortal:
```scala
val http4sV = "0.21.33"
val http4sBlazeV = "0.21.33"
```
I would think it would be preferable to just in a deterministic way add the target versions to the title rather than just have it be generic and immortal.

**Reply by BusyByte** [reply]
@rarkins  The funny thing is it's still updating both version numbers in both PRs but at least this way I can close them and have them not immortal.
